_includes/assets/js/search.js

```diff
@@ -1,115 +1,118 @@
-    if(searchValue.length) {
-      window.location.replace(that.inputElement.data('pageurl') + searchValue);
+    var searchTermsToUse = searchTerms;
+    // This is to allow for searching by indicator with dashes.
+    if (searchTerms.split('-').length == 3 && searchTerms.length < 15) {
+      // Just a best-guess check to see if the user intended to search for an
+      // indicator ID.
+      searchTermsToUse = searchTerms.replace(/-/g, '.');
     }
-  });
-
-  var escapeRegExp = function(str) {
-    return str.replace(/[\-\[\]\/\{\}\(\)\*\+\?\.\\\^\$\|]/gi, "\\$&");
-  };
-
-  if($('#main-content').hasClass('search-results')) {
-
-    var results = [],
-        that = this,
-        searchString = decodeURIComponent(location.search.substring(1)).replace("q=", "");
-
-    // we got here because of a redirect, so reinstate:
-    this.inputElement.val(searchString);
-
-    $('#main-content h1 span').text(searchString);
-    $('#main-content h1').show();
-
-    this.indicatorDataStore.getData().then(function(data) {
-
-      that.processData(data);
-
-      var searchResults = _.filter(that.indicatorData, function(indicator) {
-        return indicator.title.toLowerCase().indexOf(searchString.toLowerCase()) != -1 ||
-          indicator.description.toLowerCase().indexOf(searchString.toLowerCase()) != -1 ||
-          indicator.keywords.toLowerCase().indexOf(searchString.toLowerCase()) != -1;
-      });
-
-      // goal
-      //    indicators
-      // goal
-      //    indicators
-
-      _.each(searchResults, function(result) {
-        var goal = _.findWhere(results, { goalId: result.goalId }),
-            indicator = {
-              parsedTitle: result.title.replace(new RegExp('(' + escapeRegExp(searchString) + ')', 'gi'), '<span class="match">$1</span>'),
-              parsedDescription: result.description.replace(new RegExp('(' + escapeRegExp(searchString) + ')', 'gi'), '<span class="match">$1</span>'),
-              parsedKeywords: result.keywords.replace(new RegExp('(' + escapeRegExp(searchString) + ')', 'gi'), '<span class="match">$1</span>'),
-              hasKeywords: result.keywords && result.keywords.length,
-              hasDescription: result.description && result.description.length,
-              id: result.id,
-              title: result.title,
-              href: result.href,
-              status: result.status
-            };
-
-        if(!goal) {
-          results.push({
-            goalId: result.goalId,
-            goalTitle: result.goalTitle,
-            indicators: [indicator]
-          });
-        } else {
-          goal.indicators.push(indicator);
+    // Perform the search.
+    var results = searchIndex.search(searchTermsToUse);
+    var didYouMean = false;
+
+    // If we didn't find anything, get progressively "fuzzier" to look for
+    // "did you mean?" options.
+    if (!results.length) {
+      var fuzziness;
+      for (fuzziness = 1; fuzziness < 5; fuzziness++) {
```

Comment:
If you start your for loop with `for (var fuzziness = 1...`, you don't need to declare `var fuzziness` above.